templates/rag-momento-vector-index/rag_momento_vector_index/chain.py

```diff
@@ -0,0 +1,62 @@
+import os
+
+from langchain.chat_models import ChatOpenAI
+from langchain.embeddings import OpenAIEmbeddings
+from langchain.prompts import ChatPromptTemplate
+from langchain.pydantic_v1 import BaseModel
+from langchain.schema.output_parser import StrOutputParser
+from langchain.schema.runnable import RunnableParallel, RunnablePassthrough
+from langchain.vectorstores import MomentoVectorIndex
+from momento import (
+    CredentialProvider,
+    PreviewVectorIndexClient,
+    VectorIndexConfigurations,
+)
+
+API_KEY_ENV_VAR_NAME = "MOMENTO_API_KEY"
+if os.environ.get(API_KEY_ENV_VAR_NAME, None) is None:
+    raise Exception(f"Missing `{API_KEY_ENV_VAR_NAME}` environment variable.")
+
+MOMENTO_INDEX_NAME = os.environ.get("MOMENTO_INDEX_NAME", "langchain-test")
+
+### Sample Ingest Code - this populates the vector index with data
+### Run this on the first time to seed with data
+# from rag_momento_vector_index import ingest
+# ingest.load(API_KEY_ENV_VAR_NAME, MOMENTO_INDEX_NAME)
+
+
+vectorstore = MomentoVectorIndex(
+    embedding=OpenAIEmbeddings(),
+    client=PreviewVectorIndexClient(
+        configuration=VectorIndexConfigurations.Default.latest(),
+        credential_provider=CredentialProvider.from_environment_variable(
+            API_KEY_ENV_VAR_NAME
+        ),
+    ),
+    index_name=MOMENTO_INDEX_NAME,
+)
+retriever = vectorstore.as_retriever()
+
+# RAG prompt
+template = """Answer the question based only on the following context:
+{context}
+Question: {question}
+"""
+prompt = ChatPromptTemplate.from_template(template)
+
+# RAG
+model = ChatOpenAI()
+chain = (
+    RunnableParallel({"context": retriever, "question": RunnablePassthrough()})
```

Comment:
nit: no need to wrap this in RunnableParallel since it's being compose with a Runnable object (the prompt)

Reply:
Thanks @baskaryan . Fixed in https://github.com/langchain-ai/langchain/pull/12757/commits/841437ef0893e7c258b4d60172a6c26f6e5bee81.